compiler/rustc_mir_transform/src/match_branches.rs

```diff
@@ -368,6 +368,8 @@ impl<'tcx> SimplifyMatch<'tcx> for SimplifyToExp {
             return None;
         }
```

Comment:
Why does it make sense to compare the *lengths of the basic blocks*...?!?

Reply:
Because we are currently only merging simple BBs, we have not considered BBs that could potentially be merged even though they have different numbers of instructions.

Reply:
What's a "simple BB"? If both BBs have 24 instructions, how is that okay but one having 20 and the other 24 is not? The same number of instructions tells you *absolutely nothing* about what the BBs are doing...

Reply:
Sorry for the unclear expression. I mean the merging of basic blocks that only consider simple scenarios.
This is just an early bail-out. We assume that BBs with different the lengths of BBs cannot be merged.